### ISSUE TITLE
No accuracy drop from fp32 to w8a8 for mobilenet_v2 and resnet18

### ISSUE THREAD

**Issue by ValdezBalderas** [quic/aimet#3971]
I measured the accuracy of`mobilenet_v2` and `resnet18` following the AIMET instructions [here](https://quic.github.io/aimet-pages/releases/latest/quantsim/calibration.html). As you can see in the table below, there is no significant drop in accuracy when going from the original `fp32` model to w8a8. (But there is a dramatic drop for `w4a8` and `w4a4`.) Please note that I did **not** use any AIMET tools for accuracy improvement, neither PTQ or QAT. I did the evaluation on the validation split of the ImageNet 1K dataset.

1. Are those results expected? 
2. If so, would it be correct to conclude that there is no advantage in applying any of the AIMET tools, like PTQ or QAT, for precision w8a8 and higher, for these specific models?
3. And if so, can you suggest other simple Computer Vision models for which AIMET's PTQ tools would make a significant improvement for w8a8? I am keen to reproduce the advantages of using AIMET tools. Ideally small models similar to mobilenet or resnet, which don't require large compute resources.

In the following table, NA means Not Applicable, and `quant_data` refers to `QuantizationDataType`.

```
model          label       param_bw  output_bw  quant_data  accuracy

mobilenet_v2   fp32        NA        NA         NA          0.7167
mobilenet_v2   w16a16      16        16         float       0.7166
mobilenet_v2   w16a16      16        16         int         0.7167
mobilenet_v2   w8a16       8         16         int         0.7129
mobilenet_v2   w8a8        8         8          int         0.7132
mobilenet_v2   w4a8        4         8          int         0.0354
mobilenet_v2   w4a4        4         4          int         0.0055

resnet18       fp32       NA        NA         NA          0.6944
resnet18       w16a16     16        16         float       0.6944
resnet18       w16a16     16        16         int         0.6943
resnet18       w8a16      8         16         int         0.6931
resnet18       w8a8       8         8          int         0.6925
resnet18       w4a8       4         8          int         0.4952
resnet18       w4a4       4         4          int         0.0193
```

**Comment by quic-klhsieh** [comment]
@ValdezBalderas ,
1) We also observe ResNet18 performing well with W8A8 quantization out of the box. For MobileNetV2, we have seen larger deltas on the order of 1.5-3% between W8A8 quantization vs. full floating point models. Results vary depending on the data used for calibration as well as evaluation.

How much data are you using for calibrating MobileNetV2 as opposed to evaluation? If you are using the same amount of data for both, it's possible you are overfitting your quantization parameters to the evaluation set. We typically recommend using 500-1000 samples for calibration.

2) For models which are easily quantizable out of the box, it's true that there may not be any advantage to applying quantization techniques to them.

3) DeeplabV3 with MobileNet backbone can be another model for you to try - we have seen around a 2.5% gap between FP and W8A8 quantized models. But depending on what data you used to compute encodings for MobileNetV2, you might try using a smaller set of data for calibration to get a lower starting quantized accuracy which you can then improve with using PTQ.

**Comment by ValdezBalderas** [comment]
Here are some additional experiments, now with the effects of Adaround. TLDR is that for these models Adaround only helps for `w4a8`.

As for data, I split the validation set of ImageNet into 3 non-overlapping subsets: one for evaluation, one for calibration and one for Adround optimization. I am using much fewer examples for calibration and Adaround optimization than for evaluation (see below). 

**Questions**

1. Is using non-overlapping subsets for for evaluation, calibration and Adaround optimization is OK? Or should the datasets used for calibration and Adaround optimization be the same or at least overlapping?
2. Are the sizes of the datasets are OK for each of those tasks?
3. Can you suggest additional models for which Adaround give a good advantage? 


**Datasets**

Here are the specifics of the data used to compute the accuracies shown further below.

- **`resnet101`**:
  - Evaluation: **47872 examples**
  - Calibration: **1024 examples**
  - Adaround Optimization: **1024 examples**
  - Batch Size: **128**
- **`shufflenet_v2`**:
  - Evaluation: **29696 examples**
  - Calibration: **1024 examples**
  - Adaround Optimization: **1024 examples**
  - Batch Size: **1024**

**Impact of Adaround on `resnet101`**

| **Precision** | **Acc. Without Adaround** | **Acc. With Adaround** | **Improvement** |
|---------------|-----------------------|-------------------|-----------------|
| `w8a16`       | 0.7740               | 0.7740           | 0.00%           |
| `w8a8`        | 0.6631               | 0.6647           | 0.16%           |
| `w4a8`        | 0.6261               | 0.6468           | 2.07%           |
| `w4a4`        | 0.0130               | 0.0011           | -0.12%          |


**Impact of Adaround on `shufflenet_v2`**

| **Precision** | **Acc. Without Adaround** | **Acc. With Adaround** | **Improvement** |
|---------------|-----------------------|-------------------|-----------------|
| `w8a8`        | 0.6304               | 0.6328           | 0.24%           |
| `w4a8`        | 0.0207               | 0.5947           | 57.40%          |
| `w4a4`        | 0.0015               | 0.0024           | 0.09%           |

**Comment by quic-klhsieh** [comment]
1. Non-overlapping subsets for eval, calibration, and optimization is fine. Calibration and optimization datasets may or may not be same/overlapping, as long as they are representative of the distribution of the data that will be seen during evaluation.
2. The sizes you are using for calibration and optimization are good (~1000 samples for each is recommended).
3. You may also try mnasnet1_0, which we have seen Adaround giving good improvement with W8A8: https://pytorch.org/vision/main/models/generated/torchvision.models.mnasnet1_0.html#torchvision.models.mnasnet1_0. Our own data shows that resnet101 and shufflenet_v2 are quite robust to quantization so it makes sense that you see good W8A8 results even without applying PTQ.
